server/adapterManager.go

```diff
@@ -52,8 +52,14 @@ type AdapterManager struct {
 	ListCheckers map[string]adapters.Builder
 }
 
-// TODO: this implementation needs to be driven from external config instead of being
-// hardcoded
+// GetListCheckerAdapter returns a matching adapter for the serviceName and peerID. If there is no existing adapter,
+// it instantiates one, based on the provided adapter config.
+func (mgr *AdapterManager) GetListCheckerAdapter(serviceName string, peerID string, config *adapters.AdapterConfig) (adapters.ListChecker, error) {
```

Comment:
That's a good idea. Even if we don't ever use more facts as a key, it would be clearer if we create a type specifically for the key.